File: nz_tourism/types/payment.ts
import { BookingFormData } from './booking';

export interface PaymentIntent {
  id: string;
  amount: number;
  currency: string;
  status: string;
  clientSecret: string;
}

export interface Payment {
  id: string;
  stripePaymentIntentId: string;
  amount: number;
  currency: string;
  status: PaymentStatus;
  packageId: string;
  paidAt?: string;
  createdAt: string;
}

export enum PaymentStatus {
  Pending = 'Pending',
  Succeeded = 'Succeeded',
  Failed = 'Failed',
  Canceled = 'Canceled',
  Refunded = 'Refunded'
}

export interface CreatePaymentIntentRequest {
  packageId: string;
  amount: number;
  currency: string;
  bookingData: BookingFormData;
}


export interface UpdatePaymentStatusRequest {
  paymentIntentId: string;
  status: 'succeeded' | 'failed' | 'canceled';
}

export interface PaymentStatusResponse {
  paymentIntentId: string;
  status: string;
  amount: number;
  currency: string;
  paidAt?: string;
  failureReason?: string;
}

export interface StripePaymentIntentResponse {
  id: string;
  status: string;
  amount: number;
  currency: string;
  created: number;
}

File: WebApplication1/WandSky.Core/Entitiies/Payment.cs
using WandSky.Core.Entities;
using WandSky.Core.Entitiies;

namespace WandSky.Core.Entities
{
    public class Payment : BaseEntity
    {
        public Guid UserId { get; set; }
        public string StripePaymentIntentId { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public string Currency { get; set; } = "nzd";
        public PaymentStatus Status { get; set; }
        public string PackageId { get; set; } = string.Empty;
        public DateTime? PaidAt { get; set; }
        public string? FailureReason { get; set; }
        public string? Metadata { get; set; }

        // Navigation properties
        public User User { get; set; } = null!;
        public List<Booking> Bookings { get; set; } = new();
    }

    public enum PaymentStatus
    {
        Pending = 0,
        Succeeded = 1,
        Failed = 2,
        Canceled = 3,
        Refunded = 4
    }
}